### PR TITLE
[WIP][Marketplace] 799 marketplace stripe connect

### PR DESCRIPTION
**Pull request by tdmoneybanks**
This PR aims to provide the reaction application with the ability for the owner to enable stripe connect and set a transaction fee on orders placed through shops on the marketplace. This pr also allows sellers to connect their stripe account to the platform in order to collect payments.

### CODE REVIEW

File: imports/plugins/included/payments-stripe/client/checkout/stripe.js [hunk 1]
```diff
@@ -66,7 +66,8 @@ AutoForm.addHooks("stripe-payment-form", {
     const storedCard = cardData.type.charAt(0).toUpperCase() + cardData.type.slice(1) + " " + doc.cardNumber.slice(-4);
     Stripe.authorize(cardData, {
       total: Cart.findOne().cartTotal(),
-      currency: Shops.findOne().currency
+      currency: Shops.findOne().currency,
```
Comment (lcampanis) [inline imports/plugins/included/payments-stripe/client/checkout/stripe.js]: Shouldn't it be `SellerShops`? In order to take the currency of the seller shop the product belongs to?

File: imports/plugins/included/payments-stripe/server/methods/stripe.js [hunk 2]
```diff
@@ -97,22 +97,43 @@ Meteor.methods({
...
     });
 
-    const chargeObj = {
-      amount: "",
-      currency: "",
-      card: {},
-      capture: true
-    };
+    // check if this is a seller shop for destination and transaction fee logic
+    const sellerShop = sellerShops.findOne(paymentData.shopId);
```
Comment (lcampanis) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: Can't see an import for `SellerShops` at the top of the file. Looks like it should be 
```js
SellerShops.findOne(paymentData.shopId);
```

File: imports/plugins/included/payments-stripe/server/methods/stripe.js [hunk 3]
```diff
@@ -97,22 +97,43 @@ Meteor.methods({
...
+      };
+    }
 
     if (transactionType === "authorize") {
       chargeObj.capture = false;
     }
     chargeObj.card = parseCardData(cardData);
     chargeObj.amount = formatForStripe(paymentData.total);
     chargeObj.currency = paymentData.currency;
+    const stripeConnectSettings = Packages.findOne({ name: "reaction-stripe-connect" }).settings;
```
Comment (lcampanis) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: You can use `Reaction.getPackageSettings("reaction-stripe-connect");`

File: imports/plugins/included/payments-stripe/server/methods/stripeapi.js [hunk 4]
```diff
@@ -49,10 +49,11 @@ StripeApi.methods.getApiKey = new ValidatedMethod({
   validate: null,
   run() {
     const settings = Packages.findOne({ name: "reaction-stripe" }).settings;
-    if (!settings.api_key) {
+    const stripeConnectSettings = Packages.findOne({ name: "reaction-stripe-connect" }).settings;
```
Comment (lcampanis) [inline imports/plugins/included/payments-stripe/server/methods/stripeapi.js]: You can use Reaction.getPackageSettings("reaction-stripe-connect");

File: private/data/i18n/en.json [hunk 5]
```diff
@@ -659,6 +659,13 @@
           "orderTotal": "Total"
         }
       },
+      "marketplace": {
```
Comment (lcampanis) [inline private/data/i18n/en.json]: This should be removed from conflict.

File: server/methods/core/cart.js [hunk 6]
```diff
@@ -500,8 +500,22 @@ Meteor.methods({
...
+    const itemsByShop = _.groupBy(cart.items, "shopId");
+    for (var item in itemsByShop) {
+      if (!itemsByShop.hasOwnProperty(item)) {
+        continue;
+      }
+      order.items = itemsByShop[item];
+      order.shopId = item;
+      orders.push(order);
+    }
+    Logger.info("orders array", orders);
```
Comment (lcampanis) [inline server/methods/core/cart.js]: forgot logger?

File: imports/plugins/included/payments-stripe-connect/client/settings/settings.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import "./settings.html";
+
+Template.stripeConnectSettings.helpers({
+  StripeConnectPackageConfig() {
+    return StripeConnectPackageConfig;
+  },
+  packageData() {
+    return Packages.findOne({
+      name: "reaction-stripe-connect",
+      shopId: Reaction.getShopId()
```
Comment (tdmoneybanks) [inline imports/plugins/included/payments-stripe-connect/client/settings/settings.js]: the stripe connect settings are in the dashboard for owner. are sellers even allowed to edit payment gateway stuff? if so we can change it, just thought this package would only ever be edited by parent shop owner.

File: imports/plugins/included/payments-stripe/client/checkout/stripe.js (hunk 1 above)
Comment (tdmoneybanks) [inline imports/plugins/included/payments-stripe/client/checkout/stripe.js]: I believe you are correct. changing now.

File: imports/plugins/included/payments-stripe/server/methods/stripe.js (hunk 2 above)
Comment (tdmoneybanks) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: we will be looking at this again as its server side and doesnt have seller shops. any suggestions on how to tell if the shop isnt the parent shop?

File: imports/plugins/included/payments-stripe-connect/client/settings/settings.js (hunk 7 above)
Comment (lcampanis) [inline imports/plugins/included/payments-stripe-connect/client/settings/settings.js]: That's correct 👍

File: imports/plugins/included/payments-stripe-connect/server/i18n/el.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+[{
+  "language": "Greek",
+  "i18n": "el",
+  "ns": "reaction-stripe-connect",
+  "translation": {
+    "reaction-payments": {
+      "admin": {
+        "shortcut": {
+          "stripeConnectLabel": "Ταινία"
```
Comment (NathanielSchmeling) [inline imports/plugins/included/payments-stripe-connect/server/i18n/el.json]: The i18n files currently match what's in payments-stripe. I can change stripeConnectLabel and stripeConnectSettingsLabel as you suggested but wanted to double check. Also, should I make that change for stripeLabel and stripeSettingsLabel?

File: imports/plugins/included/payments-stripe-connect/client/settings/settings.js (hunk 7 above)
Comment (tdmoneybanks) [inline imports/plugins/included/payments-stripe-connect/client/settings/settings.js]: great! leaving for now:)

File: imports/plugins/included/payments-stripe-connect/server/Methods/methods.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import { Meteor } from "meteor/meteor";
```
Comment (tdmoneybanks) [inline imports/plugins/included/payments-stripe-connect/server/Methods/methods.js]: i made this change so not sure why its still on here? also updated to update Shops collection instead of client-only SellerShops

Comment (lcampanis) [inline imports/plugins/included/payments-stripe-connect/server/Methods/methods.js]: That's because git is case insensitive. Have a look see 
http://stackoverflow.com/questions/10523849/changing-capitalization-of-filenames-in-git

File: imports/plugins/included/payments-stripe-connect/server/i18n/el.json (hunk 8 above)
Comment (lcampanis) [inline imports/plugins/included/payments-stripe-connect/server/i18n/el.json]: True. Any reference to `Stripe Connect` should remain as is in all languages

File: imports/plugins/included/payments-stripe/server/methods/stripe.js (hunk 2 above)
Comment (lcampanis) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: Server side then yes, you'll need to access Shops directly as there is no SellerShops.
To get anything but the parent shop you can do something like
```js
Shops.findOne({
  $ne: {
     _id: Reaction.getShopId()
  }
});
```

Comment (lcampanis) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: Or 
```js
Shops.findOne(paymentData.shopId) !== Reaction.getShopId()
```

File: server/startup/registry/accounts.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import { Reaction } from "/server/api";
```
Comment (lcampanis) [inline server/startup/registry/accounts.js]: @tdmoneybanks I believe this file was also removed by RC recently

File: server/methods/core/cart.js [hunk 11]
```diff
@@ -511,8 +511,22 @@ Meteor.methods({
...
       throw new Meteor.Error(403, "Access Denied");
     }
-    const order = Object.assign({}, cart);
+
+    let orders = [];
+    let order = Object.assign({}, cart);
+    delete order.items;
+    delete order.shopId;
     const sessionId = cart.sessionId;
+    const itemsByShop = _.groupBy(cart.items, "shopId");
```
Comment (spencern) [inline server/methods/core/cart.js]: Order Splitting functionality such as splitting items by shop should probably be in a Separate PR to make these reviews go more smoothly.

File: imports/plugins/included/payments-stripe/server/methods/stripeapi.js [hunk 12]
```diff
@@ -46,14 +46,12 @@ StripeApi.methods.getApiKey = new ValidatedMethod({
   name: "StripeApi.methods.getApiKey",
   validate: null,
   run() {
-    const settings = Packages.findOne({
-      name: "reaction-stripe",
-      shopId: Reaction.getShopId()
-    }).settings;
-    if (!settings.api_key) {
+    const settings = Reaction.getPackageSettings("reaction-stripe").settings;
```
Comment (spencern) [inline imports/plugins/included/payments-stripe/server/methods/stripeapi.js]: I think we may still need to use the shopId to get the package settings

File: imports/plugins/included/product-detail-simple/client/components/variantList.js [hunk 13]
```diff
@@ -20,7 +20,7 @@ class VariantList extends Component {
     }
   }
 
-  handleChildleVariantClick = (event, variant) => {
+  handleChildVariantClick = (event, variant) => {
```
Comment (spencern) [inline imports/plugins/included/product-detail-simple/client/components/variantList.js]: Good find, but should be in a separate PR as it doesn't really pertain to the Stripe Connect Integration

File: imports/plugins/included/product-detail-simple/client/components/variantList.js [hunk 14]
```diff
@@ -149,7 +149,7 @@ class VariantList extends Component {
             <ChildVariant
               isSelected={this.props.variantIsSelected(childVariant._id)}
               media={media}
-              onClick={this.handleChildleVariantClick}
+              onClick={this.handleChildVariantClick}
```
Comment (spencern) [inline imports/plugins/included/product-detail-simple/client/components/variantList.js]: Separate PR for this too.

File: imports/plugins/included/shippo/server/i18n/index.js [hunk 15]
```diff
@@ -1,5 +1,4 @@
 import { loadTranslations } from "/server/startup/i18n";
-
```
Comment (spencern) [inline imports/plugins/included/shippo/server/i18n/index.js]: Why remove the line break?

File: private/data/i18n/en.json (hunk 5 above)
Comment (spencern) [inline private/data/i18n/en.json]: @lcampanis can you elaborate on this?

File: server/api/core/core.js [hunk 16]
```diff
@@ -50,7 +50,7 @@ export default {
 
   /**
    * registerTemplate
-   * registers Templates into the Tempaltes Collection
+   * registers Templates into the Templates Collection
```
Comment (spencern) [inline server/api/core/core.js]: Should be in a separate PR for typos, not associated with Stripe Connect

File: server/methods/core/cart.js [hunk 17]
```diff
@@ -511,8 +511,22 @@ Meteor.methods({
...
+    for (var item in itemsByShop) {
```
Comment (spencern) [inline server/methods/core/cart.js]: I think that splitting items by shop should also probably be a separate PR, though it does relate directly to Marketplace, it's not related to Stripe Connect

File: server/methods/core/cart.js (hunk 6 above)
Comment (spencern) [inline server/methods/core/cart.js]: Should remove logger.

**Comment by spencern** [comment]
Additionally, we should merge this branch in with `marketplace` because `marketplace-my-shops` has been merged already

File: imports/plugins/included/product-detail-simple/client/components/variantList.js (hunk 14 above)
Comment (zenweasel) [inline imports/plugins/included/product-detail-simple/client/components/variantList.js]: I believe this was for handling French variant clicks

File: imports/plugins/included/payments-stripe/server/methods/stripeapi.js (hunk 12 above)
Comment (spencern) [inline imports/plugins/included/payments-stripe/server/methods/stripeapi.js]: Scratch that, this method includes `shopId`

File: imports/plugins/included/product-variant/client/templates/products/productGrid/notice.js [hunk 18]
```diff
@@ -1,4 +1,5 @@
 import { ReactionProduct } from "/lib/api";
+import { Catalog } from "/lib/api";
```
Comment (spencern) [inline imports/plugins/included/product-variant/client/templates/products/productGrid/notice.js]: Never used, why import?

File: imports/plugins/included/product-admin/client/components/productAdmin.js [hunk 19]
```diff
@@ -15,6 +15,7 @@ import {
 import { Router } from "/client/api";
 import { TagListContainer } from "/imports/plugins/core/ui/client/containers";
 import update from "react/lib/update";
+import PublishContainer from "/imports/plugins/core/revisions/client/containers/publishContainer";
```
Comment (spencern) [inline imports/plugins/included/product-admin/client/components/productAdmin.js]: Never used, why import?

File: imports/plugins/included/payments-stripe/server/methods/stripeapi.js [hunk 20]
```diff
@@ -46,14 +46,12 @@ StripeApi.methods.getApiKey = new ValidatedMethod({
...
+    const stripeConnectSettings = Reaction.getPackageSettings("reaction-stripe-connect").settings;
+    if (!settings.api_key && !stripeConnectSettings.api_key) {
```
Comment (spencern) [inline imports/plugins/included/payments-stripe/server/methods/stripeapi.js]: We should merge these two plugins (stripe connect and stripe) or we should have them be separate, but relying on stripe-connect to exist for the stripe plugin to function properly is probably a bad pattern.

File: imports/plugins/included/payments-stripe/server/methods/stripe.js [hunk 21]
```diff
@@ -97,22 +97,45 @@ Meteor.methods({
...
-      amount: "",
-      currency: "",
-      card: {},
-      capture: true
-    };
+    // check if this is a seller shop for destination and transaction fee logic
+    const sellerShop = sellerShops.findOne(paymentData.shopId);
+
+    if (sellerShop && sellerShop.stripeConnectSettings) {
+      const chargeObj = {
```
Comment (spencern) [inline imports/plugins/included/payments-stripe/server/methods/stripe.js]: This `const chargeObj` will be scoped to the `if (sellerShop && sellerShop.stripeConnectSettings) {` block and will not be accessible outside that scope, so this code will not work like you expect it to.

**Comment by spencern** [comment]
Closing in favor of #2167
